app/code/Magento/Catalog/Controller/Adminhtml/Product/Initialization/Helper.php

```diff
@@ -90,14 +95,16 @@ public function __construct(
         StockDataFilter $stockFilter,
         \Magento\Catalog\Model\Product\Initialization\Helper\ProductLinks $productLinks,
         \Magento\Backend\Helper\Js $jsHelper,
-        \Magento\Framework\Stdlib\DateTime\Filter\Date $dateFilter
+        \Magento\Framework\Stdlib\DateTime\Filter\Date $dateFilter,
+        \Magento\Catalog\Model\Product\LinkTypeProvider $linkTypeProvider
```

Comment:
Indeed - but I guess it's currently better to be able to move forward without breaking things until a deprecation strategy is adopted.